dashboard/static/js/file_access_within_week.js

```diff
@@ -584,18 +584,30 @@ default_selection_logic_on_grade_selection = function(){
         $("#default_selection").show();
         $("#label_for_default_selection").html(REMEMBER_MY_SETTING);
     }
-}
+};
 
 update_default_selection = function(selection){
-    $.getJSON("/api/v1/courses/" + dashboard.course_id + "/set_user_default_selection?file=" + selection, function (initResult) {
-        if (initResult.default === 'fail') {
-            $("#label_for_default_selection").html(SETTING_NOT_UPDATED_MSG);
-            $('#default_selection').prop('checked', false);
-            return;
+    let data = {"file":selection};
+    // https://docs.djangoproject.com/en/2.1/ref/csrf/#acquiring-the-token-if-csrf-use-sessions-or-csrf-cookie-httponly-is-true
+    $.ajax({
+        url: "/api/v1/courses/" + dashboard.course_id + "/set_user_default_selection",
+        method: 'PUT',
+        data: JSON.stringify(data),
```

Comment:
I think you want to set the contentType here to application/json

`    contentType: "application/json; charset=utf-8",
`

Reply:
Do you know why this has to be json encoded at all Rather than just plain ol x-www-form-urlencoded? 

I'll send a diff suggestion in a sec.

Reply:
Nevermind, it *sounds* like the problem is it's hard to get the angular form to send regular application/x-www-form-urlencoded data. I'd just add this contentType here and leave it as-is. 

Othewise this jquery is sending this as 
"Content-Type: application/x-www-form-urlencoded; charset=UTF-8"

Reply:
Done updated to json content type

Reply:
I think if the format is json when it reaches backend the parsing json is better. I looked few code sample they were using json format with PUT call and felt of doing the same. i did not think much on this

Reply:
I the code having it urlencoded I just had to pass to QueryDict and it came out almost the same as this code. But I couldn't get Angular to do anything different than json easy so just gave up. Seemed about the same though.